riscv: config: Fix kabi changes due to config alterations

xmzzz opened the pull request:
resolve: https://github.com/RVCK-Project/rvck/issues/56

Some changes in openeuler_defconfig resulted in a large number of changes in KABI compared to the openEuler 24.03 LTS version, let's fix some of them first.

The core changes are as follows:
CONFIG_BUG=y
CONFIG_GENERIC_BUG=y
CONFIG_PREEMPT_NONE=y
CONFIG_FTRACE_MCOUNT_RECORD=y
CONFIG_PREEMPT_RCU=n
CONFIG_TASKS_RUDE_RCU=y
CONFIG_TASKS_RCU=n
CONFIG_FUNCTION_TRACER=y
CONFIG_FUNCTION_GRAPH_TRACER=y
CONFIG_DYNAMIC_FTRACE=y
CONFIG_DYNAMIC_FTRACE_WITH_REGS=y
CONFIG_FUNCTION_PROFILER=y
CONFIG_STACK_TRACER=y
CONFIG_SUSPEND=y
CONFIG_SUSPEND_FREEZER=y
CONFIG_PM_SLEEP=y
CONFIG_PM_SLEEP_SMP=y
CONFIG_PM_SLEEP_DEBUG=y
CONFIG_PM_GENERIC_DOMAINS_SLEEP=y
CONFIG_DMATEST=n
CONFIG_DMA_ENGINE_RAID=n

The other changes are generated by command:
make save_oedefconfig

Tested on sg2042.

oervci commented:
开始测试

oervci commented:
check patch done. log: https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/check-patch/63/consoleFull

oervci commented:
Kernel build success!

oervci commented:
kunit test done. log:https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/kunit-test/66/consoleFull

oervci commented:
Lava check done! result url: https://lava.oerv.ac.cn/scheduler/job/241

sterling-teng commented:
同意合并。

sterling-teng commented:
同意合并。

oervci commented:
开始测试

oervci commented:
check patch done. log: https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/check-patch/111/consoleFull

oervci commented:
Kernel build failed! log:https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/rvck-kernel-build/139/consoleFull

oervci commented:
kunit test done. log:https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/kunit-test/115/consoleFull

xmzzz committed:
riscv: config: Fix kabi changes due to config alterations

community inclusion
category: feature
bugzilla: https://github.com/RVCK-Project/rvck-olk/issues/76

--------------------------------

Some changes in openeuler_defconfig resulted in a large number of
changes in KABI compared to the openEuler 24.03 LTS version, let's
fix some of them first.

The core changes are as follows:
CONFIG_BUG=y
CONFIG_GENERIC_BUG=y
CONFIG_PREEMPT_NONE=y
CONFIG_FTRACE_MCOUNT_RECORD=y
CONFIG_PREEMPT_RCU=n
CONFIG_TASKS_RUDE_RCU=y
CONFIG_TASKS_RCU=n
CONFIG_FUNCTION_TRACER=y
CONFIG_FUNCTION_GRAPH_TRACER=y
CONFIG_DYNAMIC_FTRACE=y
CONFIG_DYNAMIC_FTRACE_WITH_REGS=y
CONFIG_FUNCTION_PROFILER=y
CONFIG_STACK_TRACER=y
CONFIG_SUSPEND=y
CONFIG_SUSPEND_FREEZER=y
CONFIG_PM_SLEEP=y
CONFIG_PM_SLEEP_SMP=y
CONFIG_PM_SLEEP_DEBUG=y
CONFIG_PM_GENERIC_DOMAINS_SLEEP=y
CONFIG_DMATEST=n
CONFIG_DMA_ENGINE_RAID=n

The other changes are generated by command:
make save_oedefconfig
make update_oedefconfig

Signed-off-by: Mingzheng Xing <xingmingzheng@iscas.ac.cn>
